chore: update getting started

JEverhart383 opened the pull request:
### Description

This is a pretty large PR that does the following things:

1. It updates and condenses the 'What is Knock' and 'How does Knock work' sections to give users a high-level overview of what they need to know to either use the quick start or example apps
2. Moves a few pages from the 'Getting Started' section to other sections of the docs to keep the flow focused on introducing Knock and getting them started with quickstart or example apps

I imagine we might want some discussion here, so kicking off the review process while this PR is in draft. 

Made some basic todos, which we may or may not want to treat as blockers based on everyone's preferences:

### Todos

- [x] Get some better screenshots of workflow steps, kind of pixelated
- [ ] Updated 5 min demo video
- [x] Maybe multi-lang code blocks

samseely commented:
@JEverhart383  My initial thoughts and feedback below. Overall big fan of these changes and love that we're reducing the number of pages in this section. 
* Had a note on product imagery but saw you included that in remaining TODOs. I think you may be able to remove the images under "Channels" and "Functions" headers too. For images like workflow canvas one we should try to size down a bit so they don't take up so much vertical space. 
<img width="711" alt="image" src="https://github.com/knocklabs/docs/assets/12143734/f6aac53d-5383-4f73-b5ea-dff9ae205b1b">

* I do wonder if "Going to production" should stay up here, just since I can see users getting caught on not promoting changes to production or something like that. I'm not sure it works under building in-app ui since some of what we're referencing here is in-app-agnostic. ("Knock and Postman" is the other one that crossed my mind given it's so often the thing you use to try out a devtool when you're getting started — but it's also easy to search for if you know you want it so down to just put this under "developer tools" in sidebar.)
* On "What is Knock" page, I'd vote to include "Overview" section with diagram of how knock works. Feels like an important way to visually distill some of the key concepts into a diagram people can understand. This diagram is ancient (first month of Knock) and we should update it. 
* Given the "What is Knock" page becomes an overview of concepts, I wonder if we change "Overview" page under "Concepts" in sidebar to "Glossary" which is really more what it is at this point. 
* Small thing but this layout feels a little funky to me with nested bullets like this. Maybe just bold the first line and then have the example follow? E.g. "**Batch.** Batch all of the comments on this document for one hour and then send one email with all of the activities."
<img width="730" alt="image" src="https://github.com/knocklabs/docs/assets/12143734/2419b00d-26c7-4bee-a8bb-19937abe35d4">

* We should definitely do multilang code blocks. These code examples are probably first ones most people reading docs would see, and goes a long way to let them switch to their preferred language. It also does a good job of showing them which languages we support today. 
* Under "Keep learning" let's put UI components above advanced concepts — I feel like more people will look for that but lmk if you disagree

rachael-t reviewed (commented):
Looks great! Some small things I noticed:
- Similar to Sam's feedback, I think maybe bolding channel types, workflows, and workflow categories would help in this section:
![Screenshot 2024-04-01 at 10 32 15 AM](https://github.com/knocklabs/docs/assets/54180641/bd7ae6a5-04a4-4d82-ba97-18e0dba395f2)
- It looks like under the next steps > build something section that we hyperlink to the quick start guide and then have a Sdkcard link to the same page as well. Initially reading this I assumed the Sdkcard would link me to something else, and it feels a bit repetitive having both (if we keep the sdkcard vs hyperlink, also we should update to "Quick start" vs "Quickstart" for better alignment). 
![Screenshot 2024-04-01 at 10 33 12 AM](https://github.com/knocklabs/docs/assets/54180641/ae0113f0-df97-4a69-ac7f-7cfe0b9c81a4)

cellomatt commented on content/getting-started/what-is-knock.mdx:
```diff
@@ -0,0 +1,248 @@
+
+#### Developer tools
+
+Knock is a developer-first platform, with both [environment](/concepts/environments) and [commit models](/concepts/commits). If you want to work with Knock resources in code, you can use our [Management API](/developer-tools/management-api) or [CLI](/developer-tools/knock-cli). Once you are sending notifications through Knock, you can also [use workflow run logs](/send-notifications/debugging-wor ...
```
```suggestion
Knock is a developer-first platform, with both [environment](/concepts/environments) and [commit models](/concepts/commits). If you want to work with Knock resources in code, you can use our [Management API](/developer-tools/management-api) or [CLI](/developer-tools/knock-cli). 

Once you're sending notifications through Knock, we offer observability tools like [workflow run logs](/send-notifications/debugging-workflows) (to examine all steps of workflow execution in your dashboard) and data streaming into a monitoring system like Datadog with [extensions](/integrations/extensions/overview).
```

cellomatt reviewed (commented):
This is great. I left a bunch of suggestions and nit grammar things, but overall this is a MASSIVE improvement. 🥳

JEverhart383 commented on content/getting-started/what-is-knock.mdx:
```diff
@@ -0,0 +1,248 @@
+  className="rounded-md shadow-sm"
+></Image>
+
+Your application can trigger workflows using our REST API, any one of our available SDKs (Node.js, .NET, PHP, Go, Elixir, .NET (C#), Java, or Python), or by integrating a CDP like Segment as an event source:
```
Since I included the multilang code block, I think we can remove the enumeration. I link to sdks page and point out dropdown lang switcher

JEverhart383 commented on content/getting-started/what-is-knock.mdx:
```diff
@@ -0,0 +1,248 @@
+Channels in Knock represent a specific provider you have configured to send notifications. You can include channel steps in your workflows to send notifications with the providers you already use in production.
+Knock supports the following channel types and providers:
+
+<AccordionGroup>
```
Good callout.

JEverhart383 commented on content/getting-started/what-is-knock.mdx:
```diff
@@ -0,0 +1,248 @@
+
+This helps product and marketing teams standardize on one templating system instead of using different templating languages for different providers. It also has the added benefit of lifting these messages out of your codebase so you can iterate quickly on customer communications without a developer.
+
+## Functions
```
I think I have a different take here, mainly that we'd want to discourage side quests where possible. If we link off, we may not get them back in this flow.

maybecolin replied:
I agree with Jeff for the most part here. Can we link out to the overview page of designing workflows at the end of this block though? That'll let someone new-tab the overview but continue here until later.

cellomatt replied:
> If we link off, we may not get them back in this flow.

cellomatt replied:
> If we link off, we may not get them back in this flow.
This is interesting, I guess I usually take the approach of giving the user as much choice as possible to make it easer/more-intuitive for them, but I'm certainly no UX expert. My personal preference when navigating a site is to be able to "choose your own adventure" to whatever I'm interested in, so that if something piques my interest I can find out more information.

For me this boils down to the root question I raised in another place which is "how much information do we want on this page." IMO, if we give partial information about something (like a general description of a more-complex feature), we should make it easy for a user to find out more about it.

cellomatt replied:
I guess my main concern here is, if we do something like say "you can use a batch function!" and that's exactly what a potential customer needs but we don't make it easy for them to find out exactly how the batch function works, that might not be a great experience for them

JEverhart383 replied:
With a bunch of content in the accordions, I went back and included links to most of the concepts. I think this does a good job of splitting the difference between enough info that someone could read just this and know 80% of what they need to about Knock, and easily digging into more details if they need to.

JEverhart383 commented on content/getting-started/what-is-knock.mdx:
```diff
@@ -0,0 +1,248 @@
+## Functions
+
+Each workflow can combine multiple function steps to model complex logic that creates better notification experiences.
+You can combine the following function steps with any number of channel steps to create great notifications:
```
This is a good callout. I think my take here is that we need some level of detail so that using just this page, someone can tell if Knock can do what they want. But simplification is the name of the game, so I leaned into using the accordion here. That way it's still very scannable, and users can opt into the weeds that seem important to them. Interested to here your thoughts.

JEverhart383 commented on content/getting-started/what-is-knock.mdx:
```diff
@@ -0,0 +1,248 @@
+
+Application developers have control over how these preference sets are presented to the user and which options to surface, but Knock enforces these preferences during every workflow run automatically.
+
+Let’s take a look at some examples of the different preference types:
```
Did the same with the accordion here, opt into complex explanation

maybecolin commented on content/getting-started/what-is-knock.mdx:
```diff
@@ -0,0 +1,255 @@
+section: Getting started
+---
+
+Knock is a hosted notifications engine that helps you implement notifications your users will love, without the effort of building and maintaining your own in-house notifications system.
```
Let's be consistent here and use notifications infrastructure as what Knock is.

maybecolin commented on next.config.js:
```diff
@@ -269,6 +269,26 @@ const nextConfig = {
         destination: "/sdks/:sdk/reference",
         permanent: false,
       },
+      {
```
@JEverhart383 Are these going to be permanent changes? or no? It feels like they are... if that's the case, we should update so that google recognizes is properly.

JEverhart383 replied:
@maybecolin I was mostly following the convention of the last N redirects, which are all not marked as permanent, but maybe they should be. In either case, my general preference for docs stuff has always been to do an initial deploy as a 302 and then follow-up once we're sure everything is working with a 301. I know it's not the best for SEO, but I'm overly cautious about stuff like this lol

cellomatt commented on content/getting-started/what-is-knock.mdx:
```diff
@@ -117,34 +105,43 @@ Knock supports the following channel types and providers:
 </AccordionGroup>
 
-When a channel step generates a notification as a part of a workflow that’s called a `message`, and Knock allows you to define dynamic message templates using a combination of a drag-and-drop editor and the liquid templating language.
+When a channel step generates a notification as a part of a workflow that’s called a `Message`, and Knock allows you to define dynamic message templates using a combination of a drag-and-drop editor and the Liquid templating language.
```
I know I'm being a bit pedantic (😅 ) but I think this sentence structure could still be a bit better grammar-wise:

```suggestion
A notification that is generated by a channel step as a part of a workflow is called a `Message`, and Knock allows you to define dynamic message templates using a combination of a drag-and-drop editor and the Liquid templating language.
```

JEverhart383 replied:
applied in another commit

cellomatt replied:
@JEverhart383 I think the wording is different? The "that's" is what was bugging me here

JEverhart383 replied:
@cellomatt fail on my end, didn't save that file. Should be good now

samseely commented on content/getting-started/what-is-knock.mdx:
```diff
@@ -0,0 +1,260 @@
+
+In addition to combining channel steps and function steps to create complex workflows, you can augment these steps with additional logic based on the user recipient, inputs from your application, or the status of previous workflow steps. These are called step conditions.
+
+## Step Conditions
```
```suggestion
## Step conditions
```
Title case --> sentence case

JEverhart383 commented:
Hey folks, just wanted to summarize some of the changes I've made since opening this draft to document our decisions. 

1. Re-ordered 'Learn more' section items based on GA data, and looked at GA data from pages we moved (nothing in the top 25)
2. Removed low quality screenshots, took new screenshot of workflow canvas to cut-off ~200px vertically
3. Moved details for channels, functions, and preferences into accordions and added links
4. Implemented multi-lang codeblock for workflow trigger
5. Implemented any copy-editing suggestions, and likely a bunch of other small changes

Here are a few other things we talked about in this thread. I'm thinking at this point, it may be best to ship what's in this PR and address some of these in separate PRs/tickets:

1. Rename Concepts > Overview to Concepts > Glossary: I personally think we should follow the same process we did for 'Getting Started' with all the docs sections to update & condense, so this section could be next
2. Overview diagram and 5-min demo: I think we'll want some design support here, and whatever imagery we use in the diagram should be reinforced in the video

LMK your thoughts on this approach @samseely @cellomatt @rachael-t

samseely commented:
@JEverhart383 I'm good with that. Let's just make sure action the overview stuff and 5-min demo view as think those would be good to add/update. 

Other note I mentioned to chris is we should start put border gray100 and a border radius on all docs images. cc @cjbell as he may have done this on a separate PR

But agreed on shipping what we've got here!

rachael-t commented:
@JEverhart383 thanks for that summary and all of your work on this! It looks fantastic, and I think we should definitely ship these changes now, and then we can revisit the other two items separately.

JEverhart383 commented:
Awesome ya'll, captured the next steps in this marketing task: https://www.notion.so/knockapp/9c1aa990054d4d45a5486f6f40de3cc4?v=e088f269d9644dfda13ee94e17acd3de&p=701721fbacf74d35b7b5ce5a1254c562&pm=s
